src/infrastructure/axios-utils.ts

```diff
@@ -8,10 +8,14 @@ import {
 	UnauthorizedHttpClientError,
 } from './http-client-errors';
 
-export const withAxiosErrorTranslation = async <T>(fn: () => Promise<T>) => {
+export const withAxiosErrorTranslation = async <T>(
+	fn: () => Promise<T>,
+	context?: unknown,
+) => {
 	try {
 		return await fn();
 	} catch (e: unknown) {
+		console.log('Error in withAxiosErrorTranslation', { context, error: e });
```

Comment:
I'm not sure how Datadog client works but will this log `context` in the Datadog?

Reply:
While it can work, it would be better to use `Logger` for consistency with other places.

Consider changing to the following:

```ts
getLogger().info(e, "An HTTP request to the external system failed.", context);
```

As an alternative option, you could add context to a thrown error itself, so it will be looged automatically by a global error handler: 

```ts
// axios-utils.ts
export const withAxiosErrorTranslation = async <T>(
	fn: () => Promise<T>,
	context?: unknown,
) => {
	try {
		return await fn();
	} catch (e: unknown) {
		throw translateAxiosError(e, context);
	}
};

const translateAxiosError = (error: unknown, context?: unknown): unknown => {
	if (!isAxiosError(error)) return error;

	switch (error.response?.status) {
		case HttpStatusCode.BadRequest:
			return new BadRequestHttpClientError(
				error.message,
				error.response.data,
				error,
				context,
			);
		case HttpStatusCode.Unauthorized:
			return new UnauthorizedHttpClientError(
				error.message,
				error.response.data,
				error,
				context,
			);
		case HttpStatusCode.Forbidden:
			return new ForbiddenHttpClientError(
				error.message,
				error.response.data,
				error,
				context,
			);
		case HttpStatusCode.NotFound:
			return new NotFoundHttpClientError(
				error.message,
				error.response.data,
				error,
				context,
			);
		default:
			return new HttpClientError(
				error.message,
				error.response?.data,
				error,
				context,
			);
	}
};
```

```ts
// http-client-errors.ts
import { CauseAwareError } from '../common/errors';

export class HttpClientError extends CauseAwareError {
	constructor(
		message?: string,
		readonly response?: unknown,
		readonly cause?: Error,
		context?: unknown,
	) {
		super(message, cause);
	}
}

// ...
```

Reply:
Thanks for the review! Yes - console.log statements do log the fields to Datadog. 

Thanks for the code sample - previously, I was having some trouble getting the pino logs to show up. In order to get the context field, I added it like 
```
getLogger().info(
	{ err: e, context },
	'An HTTP request to the external system failed.',
);
```
![image](https://github.com/atlassian-labs/figma-for-jira/assets/116598948/fac8d9a4-d75c-4362-9526-4cbe76a0fba3)